src/game_main.cpp

```diff
@@ -217,6 +256,138 @@ std::optional<FpsLimiter> createLimiter(const data::GameOptions& options) {
   }
 }
 
+
+std::string effectiveGamePath(
+  const StartupOptions& options,
+  const UserProfile& profile
+) {
+  using namespace std::string_literals;
+
+  if (!options.mGamePath.empty()) {
+    return options.mGamePath;
+  }
```

Comment:
Should userprofile's game path be updated if a valid game path is provided via options ?

Reply:
It would also help future scenarios where a user has shareware path added in user profile and wants to add full version path instead.

Reply:
Good question. I think there should definitely be a way for the user to switch to a different game path, switching from Shareware to registered version is a good example. But maybe this could also be done in the UI, like a button "Choose new game path" in the options menu?

Now that we have the game path browser in the UI, the command line game path looks like a feature for developers to me, where you might want to temporarily use a different path for testing, but use the one from the profile most of the time.

So for now, my suggestion would be that we don't update the profile, and I'll add an issue for the button to switch game path.

Reply:
I've already started working on choosing a different path from the options menu: https://github.com/lethal-guitar/RigelEngine/commits/game-path-switch